src/include/users/dialogs.rb

```diff
@@ -1120,16 +1121,9 @@ def EditUserDialog(what)
             uname = Convert.to_string(UI.QueryWidget(Id(:username), :Value))
             login_modified = false if login_modified && uname == "" # reenable suggestion
             if !login_modified
-              full = Convert.to_string(UI.QueryWidget(Id(ret), :Value))
-              full = Ops.get(Builtins.splitstring(full, " "), 0, "")
-              full = UsersSimple.Transliterate(full)
-              UI.ChangeWidget(
-                Id(:username),
-                :Value,
-                Builtins.tolower(
-                  Builtins.filterchars(full, UsersSimple.ValidLognameChars)
-                )
-              )
+              full_name = UI.QueryWidget(Id(ret), :Value)
+              username = Y2Users::UsernameGenerator.for(full_name)
```

Comment:
Thinking twice, maybe a class is not needed at all. A module would be enough. And in case of using a module, I see no problem with naming it as `Username`. Something like:

~~~
module Username
  def generate_from(value)
    transliterate(value)....
  end
end
~~~

And then here:

~~~
username = Yast2::Username.generate_from(full_name)
~~~

What do you think?